libs/ledger/include/ledger/upow/synergetic_contract.hpp

```diff
@@ -0,0 +1,192 @@
+#pragma once
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+#include "vm/analyser.hpp"
+#include "vm/common.hpp"
+
+#include "vm/compiler.hpp"
+#include "vm/module.hpp"
+#include "vm/vm.hpp"
+
+#include "ledger/chain/address.hpp"
+#include "ledger/chain/digest.hpp"
+#include "ledger/upow/synergetic_base_types.hpp"
+
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace fetch {
+
+class BitVector;
+
+namespace byte_array {
+class ConstByteArray;
+}  // namespace byte_array
+
+namespace math {
+
+class BigUnsigned;
+
+}  // namespace math
+
+namespace vm {
+
+class Module;
+class Compiler;
+class IR;
+struct Executable;
+struct Variant;
+
+}  // namespace vm
+
+namespace ledger {
+
+class StorageInterface;
+
+class SynergeticContract
```

Comment:
So this doesn't follow the same pattern? Inheriting from contract.hpp that is

Reply:
That is correct, this is largely because the interaction pattern is very different